code/modules/food_and_drinks/food.dm

```diff
@@ -6,7 +6,151 @@
 	volume = 50	//Sets the default container amount for all food items.
 	burn_state = FLAMMABLE
 
+	var/list/validLocations = list(/obj/structure/closet/secure_closet/freezer,/obj/machinery/smartfridge,/obj/machinery/food_cart,/obj/machinery/vending)
+
+	// food degrading
+	var/lifetime = 360 // our initial "life" in which we are freshest, and do not lose a lot of goodness
+	var/temperature = 0 // current temperature
+	var/reachTemp = 0 // temperature we want to reach, set through other external sources
+	var/meltable = FALSE // is the food gooey and meltable, like cheese
+	var/targetTemperature = 19 // our ideal temperature, in centigrade,by default it's just below "room" temp
+	var/meltingPoint = 100 // the temperature we melt at, if any
+	var/shouldBeHot = TRUE // are we a hot food?
+	var/freshness = 100 // current freshness rating of the food, negative is worse, positive is better, in percentage of 100
+	var/degradeSpeed = -0.5 // how far freshness changes per tick (something about twinkies), negative will gain freshness on incorrect temperatures, for gross wierd foods.
+	var/mouldRating = -50 // at what freshness rating do we begin to mould and decay
+	var/canRot = TRUE // can the food actually be affected by freshness (more twinkie jokes)
+	var/list/bestReagents = list() // what we spawn with, and what a fresh piece will offer
+	var/list/extraToxins = list() // extra stuff we can produce from being mouldy
+	var/initialDesc = "" // we store our own copy of the initial desc just in case food changes on spawn
+	var/ticksRotted = 0 // how much we've rotted, seperate from just checkign toxins for "poison" food, like carp
+	var/visibleRot = FALSE // are we now visibly rotting
+
+
 /obj/item/weapon/reagent_containers/food/New()
-		..()
-		pixel_x = rand(-5, 5)	//Randomizes postion slightly.
-		pixel_y = rand(-5, 5)
\ No newline at end of file
+	..()
+	pixel_x = rand(-5, 5)	//Randomizes postion slightly.
+	pixel_y = rand(-5, 5)
+	SSfood.insertFood(src)
+	if(targetTemperature >= 50) // gotcha to make sure any missed flags dont break
+		shouldBeHot = TRUE
+	if(!src.reagents)
+		canRot = FALSE // force off rotting for any non-reagent foods
+
+/obj/item/weapon/reagent_containers/food/initialize()
+
+// degrade handling
+/obj/item/weapon/reagent_containers/food/proc/updateFood()
```

Comment:
Timers are much more process intensive to use for regular ticking things.

And if SSobject can handle ticking 5000k items in a single tick (up till a tesla/singulo spawns), this can handle ticking at most 100 items with no performance load.